pkgs/development/tools/continuous-integration/github-runner/default.nix

```diff
@@ -22,11 +22,13 @@ buildDotnetModule rec {
     repo = "runner";
     rev = "v${version}";
     hash = "sha256-w5MqFIPTCAqQjdsWdscNnH2KNwUOp5SPFesyprXUvNE=";
-    # Required to obtain HEAD's Git commit hash
-    leaveDotGit = true;
+    # The git commit is read during the build:
+    # https://github.com/actions/runner/blob/22d1938ac420a4cb9e3255e47a91c2e43c38db29/src/dir.proj#L5
+    # But leaveDotGit actually breaks source hashes and should not be used,
+    # github-runner still reports the correct version after all
+    leaveDotGit = false;
```

Comment:
What about putting the hash in `src/dir.proj` in a `postFetch` that ends with `rm -rf $out/.git`?

Reply:
Agree.

Also, you broke `eval` with that.
The way `fetchFromGitHub` uses either `fetchzip` or `fetchgit` depending on whether `leaveDotGit`, `deepClone` or `fetchSubmodules` is set, expects `leaveDotGit` to be set to `true` or _unset_.
`leaveDotGit = false` does not work.

If you don't want to do the `leaveDotGit = true;` + `postFetch` thing @wegank suggested, please remove `leaveDotGit = false` entirely.

Reply:
I wanted to keep it for the comment otherwise someone in the future may wants to re-add it.

Reply:
> What about putting the hash in `src/dir.proj` in a `postFetch` that ends with `rm -rf $out/.git`?

That's seems like it would greatly complicate the update process because we would now also need add commit hash when updating. Users that want to know what commit the commit can just take the version tag and check what commit it correspond to.

It's actually not used for many things except some logging and telemetry:

```
src/Misc/dotnet-install.ps1
230:        CommitHash = $(if ($Data.Count -gt 1) { $Data[0] })

src/Runner.Common/HostContext.cs
228:            _userAgents.Add(new ProductInfoHeaderValue("CommitSHA", BuildConstants.Source.CommitHash));

src/Runner.Listener/Program.cs
83:                trace.Info($"Commit: {BuildConstants.Source.CommitHash}");

src/Runner.Listener/Runner.cs
76:                    _term.WriteLine(BuildConstants.Source.CommitHash);

src/Runner.Sdk/BuildConstants.cs
10:            public static readonly string CommitHash = "N/A";

src/Runner.Worker/Program.cs
35:                trace.Info($"Commit: {BuildConstants.Source.CommitHash}");

src/Test/L0/ConstantGenerationL0.cs
27:            Assert.True(BuildConstants.Source.CommitHash.Length == 40, $"CommitHash should be SHA-1 hash {BuildConstants.Source.CommitHash}");

src/dir.proj
6:            <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitHash" />
8:        <Message Text="Building $(Product): $(GitInfoCommitHash) --- $(PackageRuntime)" Importance="high"/>
17:            <BuildConstants Include="%20%20%20%20%20%20%20%20%20%20%20%20public static readonly string CommitHash = %22$(GitInfoCommitHash)%22%3B"/>
```

Reply:
I think @wegank meant something like:

https://github.com/NixOS/nixpkgs/blob/1a411f23ba299db155a5b45d5e145b85a7aafc42/pkgs/applications/virtualization/colima/default.nix#L27-L32

Reply:
I see. It still seems to be the issue though that it assumes the source is in a git repo because some tests fail otherwise.

Reply:
Fixed